src/decode.h

```diff
@@ -245,6 +245,11 @@ typedef struct rv_insn {
     /* store IR list */
     uint8_t opcode;
 
+    /* used by static analysis tool(rv_histogram)
+     * 0x1=rs1, 0x2=rs2, 0x4=rs3, 0x8=rd
+     */
+    uint8_t reg_used_mask;
```

Comment:
You can construct a reg_used table based for each instruction in `rv_histogram.c`, then look up the table using the opcode you get from function `rv_decode`. With this, we can maintain 'decode.c' and 'decode.h' unchanged.

Reply:
The approach makes sense and remains the performance of decoding.

Reply:
Based on your above comment, you gain performance lost. If @jserv think it is acceptable, this approach is suitable..